### ISSUE TITLE
MissingInputError when Bidirectional LSTM with dropout

### ISSUE THREAD

**Issue by alicanb** [fchollet/keras#5975]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

```Python
from keras.layers.wrappers import Bidirectional
from keras.layers import LSTM, Dense, Input
from keras.models import Model
import numpy as np
data=np.random.rand(1,5,10)
label=np.asarray([[1,0,0]])

inp = Input(shape=data.shape[1:])
x = Bidirectional(LSTM(units=32,recurrent_dropout=0.5))(inp)
x = Dense(3,activation='softmax')(x)
model = Model(input=inp, output=x)
model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])
model.fit(data, label, epochs=1)
```

Above code gives the following error, If I remove ```recurrent_dropout=0.5``` it trains without error:
```
MissingInputError: Input 0 of the graph (indices start from 0), used to compute if{}(keras_learning_phase, Elemwise{true_div,no_inplace}.0, Reshape{2}.0), was not provided and not given a value. Use the Theano flag exception_verbosity='high', for more information on this error.
 
Backtrace when that variable is created:

  File "<ipython-input-1-aa27b7e1da0a>", line 1, in <module>
    runfile('/home/bozkurt.a/DEJ/train_lstm_seq.py', wdir='/home/bozkurt.a/DEJ')
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line 866, in runfile
    execfile(filename, namespace)
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line 94, in execfile
    builtins.execfile(filename, *where)
  File "/home/bozkurt.a/DEJ/train_lstm_seq.py", line 11, in <module>
    from keras.models import Model
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/Keras-2.0.2-py2.7.egg/keras/__init__.py", line 3, in <module>
    from . import activations
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/Keras-2.0.2-py2.7.egg/keras/activations.py", line 3, in <module>
    from . import backend as K
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/Keras-2.0.2-py2.7.egg/keras/backend/__init__.py", line 61, in <module>
    from .theano_backend import *
  File "/home/bozkurt.a/miniconda2/lib/python2.7/site-packages/Keras-2.0.2-py2.7.egg/keras/backend/theano_backend.py", line 28, in <module>
    _LEARNING_PHASE = T.scalar(dtype='uint8', name='keras_learning_phase')  # 0 = test, 1 = train
```

**Comment by joelthchao** [comment]
Seems like wrapper is not taking good care of learning phase.
Dirty workaround:
```python
K.set_learning_phase(1)
model.fit(data, label, epochs=1)
```

**Comment by iskandr** [comment]
I have the same problem in Python 3 with a `Bidrectional` `LSTM` and `recurrent_dropout`.

**Comment by joelthchao** [comment]
@iskandr Fixed in #5985. Please update to Keras master branch.
